### ISSUE TITLE
How to build an explainer with models that have multiple input/outputs?

### ISSUE THREAD

**Issue by asheetal** [ModelOriented/DALEX#170]
I train my model with the following code with autoencoder combined with label classification.

```
history <- model %>% fit(
  x = list(factors1, factors2),
  y = list(factors1, labels),
  epochs = 10,
  batch_size = 128,
  verbose = 1,
  validation_split = 0.2)
```
and my predict command works with the following code
```
Y_test_hat <- predict(model, list(factors1, factors2), verbose=2)
```
What is the correct syntax to build an explainer for this? I want to focus on labels only?  Any suggestion how I should go about building an explainer for just label output and ignore the autoencoder output? Or maybe do both but just do feature_importance on just labels? My goal is to get feature_importance on labels of a multi input/output model

The following code generates an error
```
explainer_keras_dalex <- DALEX::explain(model,
                                        data = list(factors1, factors2),
                                        y = list(factors1, labels),
                                        label = "Keras Model",
                                        predict_function = function(m,x) predict(m, x, verbose=2))
Preparation of a new explainer is initiated
  -> model label       :  Keras Model 
  -> data              :   rows   cols 
  -> target variable   :  2  values 
Error in if (length(y) != nrow(data)) { : argument is of length zero

```

```

**Comment by pbiecek** [comment]
Hi,
explainer expects that the predict_function returns numeric scores.
Either a vector (for single selected class) or matrix (with cols for different class).

Here is an example for keras model for pythin
https://modeloriented.github.io/DALEXtra/reference/explain_keras.html

or for keras model in R
https://rawgit.com/pbiecek/DALEX_docs/master/vignettes/DALEX_and_keras.html

hope it helps

**Comment by asheetal** [comment]
Thanks, dalex::explain works fine if the model has single output classification. My curiosity is how do I handle multiple output model. In my model, one output is an autoencoder output, the second output is a classifier. Do I need to split the models and retrain them seperately again or is there a method to get a named output (classifier in my case)

**Comment by asheetal** [comment]
Addendum:
The following code did not generate any error but does not calculate any residual. So I think this is probably the incorrect way to pass the data to the model

```
explainer_keras_dalex <- DALEX::explain(model,
                                        data = list(factors1, factors2),
                                        y = labels,  #this is a matrix for second output only
                                        label = "Keras Model",
                                        predict_function = function(m,x) {
                                            pred <- predict(m, x, verbose=2)
                                            pred[[2]]  #here I return second output of the model
                                            })
```

**Comment by asheetal** [comment]
I fixed by passing a matrix to DALEX and doing an internal preprocessing in the predict_function and converting it back to the list becore calling the keras_predict
